sql-plugin/src/main/scala/com/nvidia/spark/rapids/Arm.scala

```diff
@@ -68,6 +69,15 @@ object Arm extends ArmScalaSpecificImpl {
     }
   }
 
+  /** Executes the provided code block and then closes the array buffer of resources */
+  def withResource[T <: AutoCloseable, V](r: mutable.Queue[T])(block: mutable.Queue[T] => V): V = {
+    try {
+      block(r)
+    } finally {
+      r.safeClose()
```

Comment:
Maybe, and the same question applies to the ArrayBuffer version which also doesn't clear.  There are cases where a closed object is still somewhat useful in that it has static fields, metadata, or some other part that doesn't require the original underlying resource to still be open.  So I'm a little torn on whether we want to force this to be involuntary that the queue/buffer is cleared when closed.

Assuming we do, do you want the arraybuffer version fixed here or in a followup?  It arguably adds a lot more risk to this PR to update the long-standing semantics of the ArrayBuffer version and would break anything that is examining metadata or some other "safe" thing on the instance after the underlying resource was freed.